src/modules/PopUp.jsx

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable react/no-array-index-key */
+/* eslint-disable react/prop-types */
+import React, { useEffect } from 'react';
+import SimpleSelect from './SimpleSelect';
+import WinningNumberRadio from './WinningNumberRadio';
+import StepBackRadio from './StepBackRadio';
+
+const PopUp = (props) => {
+  const {
+    isPopUpActive,
+    setIsPopUpActive,
+    bestScore,
+    boardSize,
+    setBoardSize,
+    setHistory,
+    winningNumber,
+    setWinningNumber,
+    stepBackButton,
+    setStepBackButton,
+  } = props;
+  const popupClassName = isPopUpActive ? 'popup blackout' : 'popup popup--hidden';
+  const bestScoreArray = bestScore.slice(bestScore.length - 10, bestScore.length);
+
+  const handleClick = (e) => {
+    if (e.target.className === 'popup blackout') setIsPopUpActive(false);
+  };
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClick);
+    return () => {
+      document.removeEventListener('mousedown', handleClick);
+    };
+  }, ['mousedown', handleClick]);
```

Comment:
see the previous comment

Reply:
Yeah, here it works perfectly well with empty array.